compiler/src/dotty/tools/dotc/transform/sjs/PrepJSInterop.scala

```diff
@@ -279,6 +279,39 @@ class PrepJSInterop extends MacroTransform with IdentityDenotTransformer { thisP
           val ctorOf = ref(jsdefn.JSPackage_constructorOf).appliedToTypeTree(tpeArg)
           ref(jsdefn.Runtime_newConstructorTag).appliedToType(tpeArg.tpe).appliedTo(ctorOf)
 
+        /* Rewrite js.dynamicImport[T](body) into
+         *
+         * runtime.dynamicImport[T](
+         *   new DynamicImportThunk { def apply(): Any = body }
+         * )
+         */
+        case Apply(TypeApply(fun, List(tpeArg)), List(body))
+            if fun.symbol == jsdefn.JSPackage_dynamicImport =>
+          val span = tree.span
+          val currentOwner = ctx.owner
+
+          assert(currentOwner.isTerm, s"unexpected owner: $currentOwner at ${tree.sourcePos}")
+
+          val cls = newNormalizedClassSymbol(currentOwner, tpnme.ANON_CLASS, Synthetic | Final,
+              List(jsdefn.DynamicImportThunkType), coord = span)
+          val constr = newConstructor(cls, Synthetic, Nil, Nil).entered
+
+          val applySym = newSymbol(cls, nme.apply, Method, MethodType(Nil, Nil, defn.AnyType), coord = span).entered
+          val newBody = transform(body).changeOwnerAfter(currentOwner, applySym, thisPhase)
+          val applyDefDef = DefDef(applySym, newBody)
```

Comment:
Is this the correct way to completely move a tree, and everything that it contains, inside the new `def apply(): Any` method?